contacts/tests.py

```diff
@@ -1,3 +1,123 @@
-# from django.test import TestCase
+import pytest
+from users.models import User, Seeker
+from apartments.models import Apartment, City
+from .models import Connection
+from django.db import transaction, IntegrityError
 
-# Create your tests here.
+
+@pytest.fixture
+def new_city():
+    city = City(cityName='Tel Aviv')
+    city.save()
+    return city
+
+
+@pytest.fixture
+def seeker_1(new_city):
+    user1 = create_new_user("t1@m.com", "test1", "test", "1995-05-05", "testing")
+    seeker = create_seeker(user1, new_city, "2020-05-05", 100, 3000, 2, 2, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def seeker_2(new_city):
+    user2 = create_new_user("t2@m.com", "test2", "test", "1995-05-05", "testing")
+    seeker = create_seeker(user2, new_city, "2020-05-05", 300, 3000, 3, 3, "Hi")
+    seeker.save()
+    return seeker
+
+
+@pytest.fixture
+def apartment_1(new_city):
+    user3 = create_new_user("t3@m.com", "test3", "test", "1995-05-05", "testing")
+    apartment = create_apartment(user3, new_city, "Hatotahim 5", 2500, 2, 3, "2020-05-05")
+    apartment.save()
+    return apartment
+
+
+@pytest.fixture
+def connection_1(seeker_1, apartment_1):
+    con = Connection(seeker=seeker_1, apartment=apartment_1)
+    con.save()
+    return con
+
+
+@pytest.mark.django_db
+def test_add_connection(connection_1):
+    assert connection_1 is not None
+    assert connection_1.seeker.base_user.email == "t1@m.com"
+    assert connection_1.apartment.owner.email == "t3@m.com"
+
+
+@pytest.mark.django_db
+def test_find_connection(connection_1):
+    temp_connection = Connection.objects.get(seeker=connection_1.seeker, apartment=connection_1.apartment)
+    assert temp_connection == connection_1
+
+
+@pytest.mark.django_db
+def test_invalid_users_connection(seeker_1, seeker_2):
+    with pytest.raises(ValueError):
+        with transaction.atomic():
+            Connection(seeker=seeker_1, apartment=seeker_2).save()
+
+
+@pytest.mark.django_db
+def test_invalid_duplicate_connection(connection_1):
+    with pytest.raises(IntegrityError):
+        with transaction.atomic():
+            Connection(seeker=connection_1.seeker, apartment=connection_1.apartment).save()
+
+
+@pytest.mark.django_db
+def test_multipile_connection(connection_1, seeker_2):
+    Connection(seeker=seeker_2, apartment=connection_1.apartment).save()
+    amount = Connection.objects.filter(apartment=connection_1.apartment).count()
+    expected = 2
+    assert amount == expected
+
+
+@pytest.mark.django_db
+def create_new_user(email, f_name, l_name, b_date, passw):
```

Comment:
You can apply the fixture logic I mentioned for those functions as well.

Reply:
As these aren't fixtures, I'm having trouble understanding what you mean. I've tried some things but to no avail, can you please give me an example?
(The fixture logic applied above uses these functions to create objects in the models)

Reply:
I meant that if you're already going the fixture road, you can make `create_new_user` into a fixture as well. fixtures can use other fixtures, so you should not have issues applying it in the current situation.